woocommerce/woocommerce-android
# [Hackweek][Backend Receipts] Remove check on dev WC core version

Closes: #10651


### Description

As the currently released version (8.7+) of WC core supports backend-generated receipts, we can remove support of the dev version from the code

Also, I noticed a small bug - "see receipt" button was no reenabled if the generation of a receipt fails.

### Testing instructions


* On a site with WC core 8.7.0+: Orders -> Paid Order -> see receipt visible (there is a reported bug in the backend that in some cases receipts are not generated)
* On a site with WC core 8.7.0-: Orders -> Paid Order -> see receipt not visible

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/4923871/a8fd815d-2209-4c5f-92d6-20787430bfdc
